reproducible/nodebootstrap.bzl

```diff
@@ -0,0 +1,92 @@
+# Copyright 2017 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rule for building nodebootstrap rootfs tarballs."""
+
+def _impl(ctx):
+    # Strip off the '.tar'
+    image_name = ctx.attr._builder_image.label.name.split('.', 1)[0]
+    # docker_build rules always generate an image named 'bazel/$package:$name'.
+    builder_image_name = "bazel/%s:%s" % (ctx.attr._builder_image.label.package,
+                                          image_name)
+
+    # Generate a shell script to run the build.
+    build_contents = """\
+#!/bin/bash
+set -ex
+
+docker load -i {0}
+# Run the builder image.
+cid=$(docker run -d --privileged {1})
+docker attach $cid
+# Copy out the rootfs.
+docker cp $cid:/workspace/rootfs.tar.gz {2}
```

Comment:
maybe name this something more relevant? It's not really a rootfs.

Reply:
done